src/Microsoft.Extensions.Options.Validation/OptionsValidationServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using Microsoft.Extensions.Options.Validation;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for adding options services to the DI container.
+    /// </summary>
+    public static class OptionsValidationServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds services required for using options validation.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
+        /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+        public static IServiceCollection AddOptionsValidation(this IServiceCollection services)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            services.AddSingleton<IOptionsValidatorManager, OptionsValidatorManager>();
+
+            return services;
+        }
+
+        public static IServiceCollection Validate<TOptions>(this IServiceCollection services, Expression<Func<TOptions, bool>> validateExpression)
```

Comment:
What do you think about a less restrictive signature for the validation?  If we call it Validation but treat it effectively as a second set of actions that run after the Configure step.  This would let you write Validations that both Assert/SetDefaults.  What you have today I think would be good as something built on top, sugar that makes it easy for you to register an expression that returns a bool. 

Something like `Validate(Action<TOptions> options)` as well.

What do you think?